src/Databases/DataLake/RestCatalog.cpp

```diff
@@ -591,9 +591,16 @@ bool RestCatalog::getTableMetadataImpl(
     std::string location;
     if (result.requiresLocation())
     {
-        location = metadata_object->get("location").extract<String>();
-        result.setLocation(location);
-        LOG_TEST(log, "Location for table {}: {}", table_name, location);
+        if (metadata_object->has("location"))
+        {
+            location = metadata_object->get("location").extract<String>();
+            result.setLocation(location);
+            LOG_TEST(log, "Location for table {}: {}", table_name, location);
+        }
+        else if (!result.isLightweight())
```

Comment:
Why do we need this? There is already a check `requiresLocation`, so if location is not needed, then just don't call `withLocation`